src/lcm/typing.py

```diff
@@ -1,8 +1,27 @@
 from enum import Enum
-from typing import Any, Protocol
+from typing import Any, Generic, Protocol, TypeVar
 
 from jax import Array
 
+T = TypeVar("T")
+
+
+class Param(Generic[T]):
+    def __init__(self, value: T):
+        self.value = value
+
+    def get(self) -> T:
+        return self.value
+
+
+type ContinuousState = Array
+type ContinuousAction = Array
+type DiscreteState = Array
+type DiscreteAction = Array
+type AuxiliaryVariable = Array
```

Comment:
```suggestion
type DerivedBool = Bool[Array, "..."]
```
using `jaxtyping`. And add `Int` and `Float` versions. Without the type info, the hint is not quite as helpful. `ContinuousState` should become `Float[Array, "..."]`  and `DiscreteState` probably `Int[Array, "..."]` (though I am not perfectly sure about the dataclasses here). 

We might be able to do better than the `"..."` for the shape info ([link](https://docs.kidger.site/jaxtyping/api/array/#shape)), but I am too far removed of lcm's internals right now to tell.

Reply:
Note I kind of like the `Derived` better than `Auxiliary` because it makes clear that the stuff is somehow based on states and actions. But no need to spend time on changing the entire naming conventions if it takes too much time / puts knots in your head.